src/overseer/store/jdbc.clj

```diff
@@ -0,0 +1,238 @@
+       (map :job_id)
+       set))
+
+(defn dependents [db-spec job-id]
+  "Returns the set of all Job IDs that depend upon given `job-id`, both
+  directly and transitively"
+  (loop [all-dependents #{}
+         visited #{}
+         to-visit #{job-id}]
+    (if (empty? to-visit)
+      (set (map #(:job/id (query-job db-spec %)) all-dependents))
+      (let [job-id' (first to-visit)
+            dependents (direct-dependents db-spec job-id')
+            all-dependents' (set/union all-dependents dependents)
+            visited' (conj visited job-id')
+            to-visit' (set/difference
+                        (set/union to-visit dependents)
+                        visited')]
+        (recur all-dependents' visited' to-visit')))))
+
+(defn job-dep-jdbc-maps
+  "Return a seq of dependency rows to insert for a Graph<Job>"
+  [graph]
+  (let [nodes (loom.graph/nodes graph)
+        edges (loom.graph/edges graph)
+
+        dep-edges ; Basically recreate adjacency list {job [dep-jobs]}
+        (std/map-kv (partial map second) (group-by first edges))
+
+        job->dep-maps
+        (fn [job]
+          (when (contains? dep-edges job)
+            (->> (get dep-edges job)
+                 (map (fn [dep]
+                        {:job_id (:job/id job)
+                         :dep_id (:job/id dep)})))))]
+    (->> nodes
+         (mapcat job->dep-maps)
+         (filter identity))))
+
+(def default-table-opts
+  "Backend-specific system configuration options"
+  {:mysql
+   {:table-spec "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC"}})
+
+(defn install' [adapter db-spec]
+  (j/db-do-commands
+    db-spec
+    [(j/create-table-ddl
+       :overseer_jobs
+       [[:id "VARCHAR(64) PRIMARY KEY"]
+        [:type "VARCHAR(255)"]
+        [:args "VARCHAR(2048)"]
+        [:status "VARCHAR(255)"]
+        [:failure "VARCHAR(2048)"]
+        [:heartbeat :int]
+        [:lock_version "INT NOT NULL DEFAULT '0'"]]
+       (get default-table-opts adapter {}))
+     (j/create-table-ddl
+       :overseer_dependencies
+       [[:job_id "VARCHAR(64)"]
+        [:dep_id "VARCHAR(64)"]]
+       (get default-table-opts adapter {}))
+     "CREATE INDEX index_overseer_jobs_on_status ON overseer_jobs (status)"
+     "CREATE INDEX index_overseer_dependencies_on_job_id ON overseer_dependencies (job_id)"
+     "CREATE INDEX index_overseer_dependencies_on_dep_id ON overseer_dependencies (dep_id)"])
+  :ok)
+
+(defn- jobs-with-unfinished-dep
+  "Return the seq subset of `job-ids` that have an unfinished direct dependency"
+  [db-spec job-ids]
+  (if (seq job-ids)
+    (->> (j/query
+           db-spec
+           (flatten [(str "SELECT job_id FROM overseer_dependencies "
+                          "INNER JOIN overseer_jobs ON overseer_jobs.id = overseer_dependencies.dep_id "
+                          (format "WHERE overseer_dependencies.job_id IN (%s) " (in-placeholders job-ids))
```

Comment:
The thinking was that callers would just slot the installation into their existing DB, rather than creating a separate one in all cases (and then of course we don't want to clobber names). This is how DelayedJob works in Ruby, for example.